core/modules/sql/testSqlTransaction.cc

```diff
@@ -0,0 +1,260 @@
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "SqlTransaction.h"
+
+// System headers
+#include <iostream>
+#include <string>
+#include <unistd.h> // for getpass
+
+// Third-party headers
+
+// Qserv headers
+#include "sql/SqlConnection.h"
+#include "sql/SqlErrorObject.h"
+#include "sql/SqlResults.h"
+
+// Local headers
+
+// Boost unit test header
+#define BOOST_TEST_MODULE SqlTransaction_1
+#include "boost/test/included/unit_test.hpp"
+
+using lsst::qserv::mysql::MySqlConfig;
+using namespace lsst::qserv::sql;
+
+#define DB_NAME "testSqlTransX675sdrt"
+#define TABLE_NAME "TEST123"
+#define FULL_TABLE_NAME DB_NAME "." TABLE_NAME
+
+namespace {
+
+
+struct TestDBGuard {
+    TestDBGuard() {
+        sqlConfig.hostname = "";
+        sqlConfig.port = 0;
+        sqlConfig.username = "root";
+        sqlConfig.password = getpass("Enter mysql root password: ");
+        std::cout << "Enter mysql socket: ";
+        std::cin >> sqlConfig.socket;
+        sqlConfig.dbName = DB_NAME;
+
+        // need config without database name
+        MySqlConfig sqlConfigLocal = sqlConfig;
+        sqlConfigLocal.dbName = "";
+        SqlConnection sqlConn(sqlConfigLocal);
+
+        SqlErrorObject errObj;
+
+        // create database
+        sqlConn.createDb(DB_NAME, errObj);
+    }
+
+    ~TestDBGuard() {
+        SqlConnection sqlConn(sqlConfig);
+        SqlErrorObject errObj;
+        sqlConn.dropDb(sqlConfig.dbName, errObj);
+    }
+
+    MySqlConfig sqlConfig;
+};
+
+}
+
+struct PerTestFixture {
+    PerTestFixture() {
+        sqlConn = std::make_shared<SqlConnection>(testDB.sqlConfig);
+
+        // create table (must be InnoDB)
+        std::string query = "CREATE TABLE " FULL_TABLE_NAME " (X INT, Y INT) ENGINE=InnoDB";
+        SqlErrorObject errObj;
+        sqlConn->runQuery(query, errObj);
+    }
+    ~PerTestFixture() {
+        // drop table
+        std::string query = "DROP TABLE " FULL_TABLE_NAME;
+        SqlErrorObject errObj;
+        sqlConn->runQuery(query, errObj);
+    }
+
+    static TestDBGuard testDB;
+    std::shared_ptr<SqlConnection> sqlConn;
```

Comment:
Nit: Should this be unique_ptr instead?  It looks like the pointer-smartness is only being used for lifetime control (pointer is deref'd everywhere below before being passed, unless I missed somthing?)

For that matter, could this just be an embedded member var instead of a pointer?